### ISSUE TITLE
Fluent api for routing allows weird combos

### ISSUE THREAD

**Issue by johnsimons** [Particular/NServiceBus#3721]
Eg.
```
endpointConfiguration.UseTransport<RabbitMQTransport>().UnicastRouting().UnicastRouting().UnicastRouting();
```

**Comment by SimonCropp** [comment]
General rule for fluent apis: if a method returns the same type it extends then the fluent api is broken

**Comment by SeanFeldman** [comment]
2nd that. `ExposeSettings` has `.UnicastRouting()` that makes it very strange.

`endpointConfiguration.UseTransport<AzureServiceBusTransport>().UnicastRouting()`

Which is polluting the API.

**Comment by bording** [comment]
This is fixed by #3816.
